scripts/cmake-configurator/parser/cmake_row.py

```diff
@@ -75,7 +78,13 @@ def serialize(self) -> str:
       returns serialized version of variable. Expects to be called only on
       variable types.
       """
-      return self.name + ":" + self.var_type + "=" + self.val
+      serialized_val = self.val
```

Comment:
**STYLE:** this looks a bit cumbersome. You could re-write it as:
```
if type == "BOOL":
    ser_val = "ON" if self.val else "OFF"
else:
    ser_val = self.val
```

Also, it's really **NOT OK** even if Python scripts to use strings for descripting types. `var_type` should be an Enum as well.

Reply:
Should we use strings here? it makes more sense for the final serialization

Reply:
Maybe I'm confused. What self.val is supposed to contain? I saw it might contain the actual bool value in case the type is BOOL, right? In case the type is something else, like integer, shouldn't it contain an integer instead of a string representing it? Of course if the type is STRING, then it will contain a string value. 

The idea of serialization is to convert the in-memory representation object to a flat-string. So, from self.val, we "compute" seralized_val, right? Actually my suggested code is wrong as well, because it takes self.val as it is in all the other cases and that doesn't look right.

Reply:
Yeah, that's right. Right now the only `var_type` that actually needs to change its `var_value` is `BOOL`, so i think your code would actually work, but it might need to be extended in the future. The only thing that needs to be changed from your code is  
```python
      if self.var_type == var_type.BOOL:
````